decentraland/unity-renderer
# fix: highlight backpack hair selection

## What does this PR change?
Fixes #2908 

`ImageColorOnToggle` Component on all Toggles inside of CategorySelection of the AvatarEditorHUB was causing this issue. Removing it ix the problem. I also refactored UIToggle to replace inheritance via composition.

## How to test the changes?

1. Open backpack
2. Switch to Head and choose there some sub-category other than Hairs
3. Switch to another category (Skins, as example)
4. Switch back to Head
5. Hair sub-category should stay black and sub-category chosen on step 2 

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
